flink-connectors/flink-connector-kinesis/pom.xml

```diff
@@ -193,25 +193,30 @@ under the License.
 							<artifactSet combine.children="append">
 								<includes>
 									<include>com.amazonaws:*</include>
+									<include>com.fasterxml:*</include>
 									<include>com.google.protobuf:*</include>
 									<include>org.apache.httpcomponents:*</include>
 									<!-- Java 11 specific inclusion; should be a no-op for other versions -->
 									<include>javax.xml.bind:jaxb-api</include>
 								</includes>
 							</artifactSet>
 							<relocations combine.children="override">
-								<!-- DO NOT RELOCATE GUAVA IN THIS PACKAGE -->
 								<relocation>
-									<pattern>com.google.protobuf</pattern>
-									<shadedPattern>org.apache.flink.kinesis.shaded.com.google.protobuf</shadedPattern>
+									<pattern>com.amazonaws</pattern>
+									<shadedPattern>org.apache.flink.streaming.kinesis.shaded.com.amazonaws</shadedPattern>
 								</relocation>
 								<relocation>
-									<pattern>com.amazonaws</pattern>
-									<shadedPattern>org.apache.flink.kinesis.shaded.com.amazonaws</shadedPattern>
+									<pattern>com.fasterxml</pattern>
+									<shadedPattern>org.apache.flink.streaming.kinesis.shaded.com.fasterxml</shadedPattern>
+								</relocation>
+								<!-- DO NOT RELOCATE GUAVA IN THIS PACKAGE -->
```

Comment:
Why is guava not included ?

Reply:
Not sure; it's not justified in the commit, PR or JIRA.

Without further research my guess would be that guava is exposed in one of the Kinesis APIs, and we wanted to ensure that users don't try to use a different guava version. (If it were shaded it wouldn't show up in the pom, and working against relocated namespaces i kind of a pain). But this is purely an optimistic guess.

Reply:
That seems plausible. @tzulitai or @tweise do you remember the reason for this? Is this still valid?

Reply:
I see it used in `KinesisProducer`, but not on the API surface. We should be able to use flink shaded in this case, since we already depend on it via `flink-streaming-java`:
```
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ flink-connector-kinesis_2.11 ---
[INFO] org.apache.flink:flink-connector-kinesis_2.11:jar:1.11-SNAPSHOT
[INFO] +- org.apache.flink:flink-streaming-java_2.11:jar:1.11-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-core:jar:1.11-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-runtime_2.11:jar:1.11-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-clients_2.11:jar:1.11-SNAPSHOT:provided
[INFO] |  |  \- org.apache.flink:flink-optimizer_2.11:jar:1.11-SNAPSHOT:provided
[INFO] |  +- org.apache.flink:flink-shaded-guava:jar:18.0-9.0:provided
[INFO] |  \- org.apache.commons:commons-math3:jar:3.5:provided
[INFO] +- com.google.guava:guava:jar:18.0:compile

```

Reply:
`KinesisProducer#addUserRecord` exposes guava in the API since it returns a `ListenableFuture`.

So relocating guava is out. I'll amend the pom to mention this explicitly.